examples/stm32/f0/other/can/can.c

```diff
@@ -0,0 +1,182 @@
+#define BTR500k 0x001c0005
+
+struct can_tx_msg {
+	uint32_t std_id;
+	uint32_t ext_id;
+	uint8_t ide;
+	uint8_t rtr;
+	uint8_t dlc;
+	uint8_t data[8];
+};
+
+struct can_rx_msg {
+	uint32_t std_id;
+	uint32_t ext_id;
+	uint8_t ide;
+	uint8_t rtr;
+	uint8_t dlc;
+	uint8_t data[8];
+	uint8_t fmi;
+};
+
+struct can_tx_msg can_tx_msg;
+struct can_rx_msg can_rx_msg;
+
+
+static void systick_setup(void)
+{
+	/* 72MHz / 8 => 9000000 counts per second */
+	systick_set_clocksource(STK_CSR_CLKSOURCE_AHB);
+
+	/* 9000000/9000 = 1000 overflows per second - every 1ms one interrupt */
+	/* SysTick interrupt every N clock pulses: set reload to N-1 */
+	systick_set_reload(8999);
+
+	systick_interrupt_enable();
+
+	/* Start counting. */
+	systick_counter_enable();
+}
+
+static void can_setup(void)
+{
+	/* Enable peripheral clocks. */
+
+	//B8 Rx
+	//B9 Tx
+	rcc_periph_clock_enable(RCC_GPIOB);
+
+	rcc_periph_clock_enable(RCC_CAN);
+
+	/* Configure CAN pin: RX Alternate Function (input pull-up). */
+	gpio_mode_setup(GPIOB, GPIO_MODE_AF, GPIO_PUPD_PULLUP, GPIO8);
+	gpio_set_af(GPIOB, GPIO_AF4, GPIO8);
+
+	/* Configure CAN pin: TX Alternate Function */
+	gpio_mode_setup(GPIOB, GPIO_MODE_AF, GPIO_PUPD_NONE, GPIO9);
+	gpio_set_af(GPIOB, GPIO_AF4, GPIO9);
+
+	/* Reset CAN. */
+	can_reset(CAN1);
+
+	/* CAN cell init. */
+	if (can_init(CAN1,
+		     false,           /* TTCM: Time triggered comm mode? */
+		     true,            /* ABOM: Automatic bus-off management? */
+		     false,           /* AWUM: Automatic wakeup mode? */
+		     true,           /* NART: No automatic retransmission? */
+		     false,           /* RFLM: Receive FIFO locked mode? */
+		     false,           /* TXFP: Transmit FIFO priority? */
+		     SJWFROMBTRVAL(BTR500k),
+			 TS1FROMBTRVAL(BTR500k),
+		     TS2FROMBTRVAL(BTR500k),
+			 BRPFROMBTRVAL(BTR500k),
+		     false,				 /* Loopback */
+		     false))             /* Silent */
+	{
+		/* can init returned a non zero value
+		 * this can be caused by wrong configuration of the alternate function pins,
+		 * or other reasons keeping the can module from reading 11 recessive bits for
+		 * leaving initialization phase
+		 */
+		__asm__("bkpt");
+	}
+	/* CAN filter 0 init. */
+	can_filter_id_mask_32bit_init(CAN1,
+				0,     /* Filter ID */
+				0,     /* CAN ID */
+				0,     /* CAN ID mask */
+				0,     /* FIFO assignment (here: FIFO0) */
+				true); /* Enable the filter. */
+
+	/* NVIC setup. */
+		nvic_enable_irq(NVIC_CEC_CAN_IRQ);
+		nvic_set_priority(NVIC_CEC_CAN_IRQ, 1);
+
+	/* Enable CAN RX interrupt. */
+	can_enable_irq(CAN1, CAN_IER_FMPIE0);
+}
+
+
+uint8_t rxcnt=0;
+
+void sys_tick_handler(void)
+{
+	static int temp32 = 0;
+	static uint8_t data[8] = {0, 1, 2, 0, 0, 0, 0, 0};
+
+	/* We call this handler every 1ms so 1000ms = 1s on/off. */
+	if (++temp32 != 1000)
+		return;
+
+	temp32 = 0;
+
+	/* Transmit CAN frame. */
+	data[0]++;
+
+	data[7]=rxcnt; /* send count of received messages in last byte */
+	can_transmit(CAN1,
```

Comment:
So could you please be that person?  continuing to copy/paste bad code just makes it even harder for someone to do it in the future.